grease/src/Grease/Macaw/ResolveCall.hs

```diff
@@ -124,8 +245,11 @@ lookupFunctionHandle ::
   -- | Map of names of overridden functions to their implementations
   Map.Map W4.FunctionName (MacawFunctionOverride p sym arch) ->
   ErrorSymbolicFunCalls ->
+  LookupFunctionHandleDispatch p sym arch ->
```

Comment:
It looks like this function mostly computes a `LookupFunctionHandleResult`, and then applies `dispatch` to it. Perhaps it would be more clear to factor out a function that explicitly returns a LFHR, and replace the body of this function with something that just calls that one and applies `dispatch` to the result?